### ISSUE TITLE
WeatherProperty:SkyTemperature shifts schedules after March 1

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4866]
When using WeatherProperty:SkyTemperature with a schedule, the sky temperatures line up with the specified schedule up through Feb 28 or March 1, then it repeats a day and is a day behind from there on out to the end of the year.  When using schedule:file, the output for March 1 is a repeat of Feb 28.  When using schedule:compact, the output for March 2 is a repeat of March 1.  Either way, WeatherProperty:SkyTemperature seems to be making an adjustment for leap day that is not consistent with the schedule manager.

**Comment by mjwitte** [comment]
@rglazer2 I think I found the problem.  When schedules are assembled, the arrays are always set up for 366 days.  So, when accessing a schedule, the leap year flag is essentially always true.  This can be seen here in WeatherManager::SetCurrentWeather (line 2020):

```
        if ( HourOfDay == 1 ) {
            DayOfYear_Schedule = JulianDay( Month, DayOfMonth, 1 );
        }
```

The main schedule function GetCurrentScheduleValue 

Note here that JulianDay is always called with a 1 for the leapyear adder.

In GetScheduleValuesForDay, if JDay is not present, it uses DayOfYear_Schedule

```
        // Determine which Week Schedule is used
        if ( ! present( JDay ) ) {
            WeekSchedulePointer = Schedule( ScheduleIndex ).WeekSchedulePointer( DayOfYear_Schedule );
        } else {
            WeekSchedulePointer = Schedule( ScheduleIndex ).WeekSchedulePointer( JDay );
        }
```

When GetScheduleValuesForDay is called, TomorrowVariables.DayOfYear is passed as JDay:
`GetScheduleValuesForDay( WPSkyTemperature( Environment( Environ ).WP_Type1 ).SchedulePtr, TomorrowSkyTemp, TomorrowVariables.DayOfYear, CurDayOfWeek );`

That is set in WeatherManager, line 2667
`TomorrowVariables.DayOfYear = JulianDay( WMonth, WDay, LeapYearAdd );`

So, I would add a new variable named TomorrowVariables.DayOfYear_Schedule and set it to this:
`TomorrowVariables.DayOfYear_Schedule = JulianDay( WMonth, WDay, 1);`
And change the argument to GetScheduleValuesForDay where it's setting the sky temps (3 occurences).

There are three other uses of TomorrowVariables.DayOfYear in WeatherManager:

```
                        CalculateDailySolarCoeffs( TomorrowVariables.DayOfYear, A, B, C, AVSC, TomorrowVariables.EquationOfTime, TomorrowVariables.SinSolarDeclinAngle, TomorrowVariables.CosSolarDeclinAngle );
                        if ( CurDayOfWeek <= 7 ) {
                            CurDayOfWeek = mod( CurDayOfWeek, 7 ) + 1;
                        }
                        TomorrowVariables.DayOfWeek = CurDayOfWeek;
                        TomorrowVariables.DaylightSavingIndex = DSTIndex( TomorrowVariables.DayOfYear );
                        TomorrowVariables.HolidayIndex = SpecialDayTypes( TomorrowVariables.DayOfYear );
```

I'm fairly certain the CalculateDailySolarCoeffs is OK.  And looking at SetDSTDateRanges() and SetSpecialDayDates() it looks like TomorrowVariables.DaylightSavingIndex and .HolidayIndex are OK as-is.

**Comment by mjwitte** [comment]
@rglazer2 There is one other suspicious use of GetScheduleValuesForDay.
[SystemAvailabilityManager](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/SystemAvailabilityManager.cc#L2255) when checking the fan schedule for tomorrow.  I will try to come up with a defect file that shows if this is broken.

**Comment by mjwitte** [comment]
@rglazer2 Tried to prove the schedules were broken using variations of OptimalStart_RefBldgLargeOfficeNew2004_Chicago.idf, but so far it looks OK.  So we should leave that code as-is for now.

**Comment by rglazer2** [comment]
@mjwitte Thanks. Your suggestion fixed the problem with sky temp. I will create the pull request now.

**Comment by Myoldmopar** [comment]
Fixed with #5114
